### DISCUSSION TITLE
PrimeNG table: row selection not working

### DISCUSSION THREAD

**Discussion by ghost** [primefaces/community#2928] [PrimeNG] [answered]
The table works: headers fill, rows fill with data, columns sort. Single row selection doesn't work. 

p-table:
```
<p-table [value]="releases" [columns]="headers" responsiveLayout="scroll"
    [paginator]="true" [rows]="pageRows" dataKey="index" 
    (onRowSelect)="onRowSelect($event)"  selectionMode="single"  
    [(selection)]="selectedRelease" [resizableColumns]="true" sortMode="multiple">
```

onRowSelect:
```
  // onRowSelect method
  onRowSelect(event: any) {
    this.messageService.add({ severity: 'info', summary: 'Release Selected', detail: event.data.name });
  }
```

The function is called, but when breaking at the message call the event.data and event.index are undefined. This worked before an Angular/PrimeNG upgrade. Angular 17.3, PrimeNG 17.16.1

**Comment by qwadrox** [comment]
@kvea2git Hi!

I checked it and `event.data` works correctly it is not undefined. Only `event.index` is undefined in these examples. 

latest: 
https://stackblitz.com/edit/wuh4mm?file=src%2Fapp%2Ftable-selection-events-demo.ts

17: 
https://stackblitz.com/edit/ukzgrp?file=tsconfig.json,src%2Fapp%2Ftable-single-selection-demo.ts


but it's because in the example pSelectableRowIndex is not used.  Here is how it should be used: 


```html
<ng-template pTemplate="body" let-product let-rowIndex="rowIndex">
  <tr [pSelectableRow]="product" [pSelectableRowIndex]="rowIndex">
    <td>{{product.code}}</td>
    <td>{{product.name}}</td>
    <td>{{product.category}}</td>
    <td>{{product.quantity}}</td>
  </tr>
</ng-template>
```

We have to use ` let-rowIndex="rowIndex"` on the body template and then pass it as `[pSelectableRowIndex]="rowIndex"`

@kvea2git  did you add these? 
If yes and still not working could you create a sample?

**Reply by qwadrox** [reply]
In old docs this was mentioned in an example: 
https://github.com/qwadrox/primeng/blob/f0b6fc98d08285a61e63a0200cd410f75f195414/src/app/showcase/components/table/tabledemo.html#L1354-L1367

**Reply by ghost** [reply]
Thanks for the tip on pSelectableRowIndex.

**Comment by ghost** [comment, marked as answer]
I had omitted the iterator variable on the body template. I had

```
<ng-template pTemplate="body" let-release>
  <tr [pSelectableRow]>
```
where I needed

```
  <ng-template pTemplate="body" let-release>
    <tr [pSelectableRow] = "release">
```

I should have put more of the p-table code on the question. This did work before the framework upgrade. I edited it for other reasons and introduced this. Thanks anyway
